python/cudf/cudf/core/column/categorical.py

```diff
@@ -582,7 +595,26 @@ def is_monotonic_decreasing(self):
         return self._is_monotonic_decreasing
 
     def as_categorical_column(self, dtype, **kwargs):
-        return self
+        if isinstance(dtype, str) and dtype == "category":
+            return self
+        if isinstance(
+            dtype, (cudf.core.dtypes.CategoricalDtype, pd.CategoricalDtype)
+        ):
+            if (dtype.categories is None) and (dtype.ordered is None):
+                return self
+
+        dtype = CategoricalDtype(
+            categories=dtype.categories, ordered=dtype.ordered
+        )
```

Comment:
Should we add a shallow copy mechanism to `CategoricalDtype` to avoid having to reconstruct this like when a shallow copy is needed?

Reply:
So the case here is different, I actually wanted to convert `pd.CategoricalDtype` to `cudf.CategoricalDtype`, to avoid confusion now creating `CategoricalDtype` only if it is a `pd.CategoricalDtype`.